src/application/F3DOptionsParser.cxx

```diff
@@ -179,101 +174,101 @@ class ConfigurationOptions
+    this->DeclareOption(grp4, "bar", "b", "Show scalar bar", options.getAsBoolRef("bar"));
+    this->DeclareOption(grp4, "colormap", "", "Specify a custom colormap", options.getAsDoubleVectorRef("colormap"), true, "<color_list>");
+    this->DeclareOption(grp4, "volume", "v", "Show volume if the file is compatible", options.getAsBoolRef("volume"));
+    this->DeclareOption(grp4, "inverse", "i", "Inverse opacity function for volume rendering", options.getAsBoolRef("inverse"));
 
     auto grpCamera = cxxOptions.add_options("Camera");
-    this->DeclareOption(grpCamera, "camera-position", "", "Camera position", options.CameraPosition, false, true, "<X,Y,Z>");
-    this->DeclareOption(grpCamera, "camera-focal-point", "", "Camera focal point", options.CameraFocalPoint, false, true, "<X,Y,Z>");
-    this->DeclareOption(grpCamera, "camera-view-up", "", "Camera view up", options.CameraViewUp, false, true, "<X,Y,Z>");
-    this->DeclareOption(grpCamera, "camera-view-angle", "", "Camera view angle (non-zero, in degrees)", options.CameraViewAngle, false, true, "<angle>");
-    this->DeclareOption(grpCamera, "camera-azimuth-angle", "", "Camera azimuth angle (in degrees)", options.CameraAzimuthAngle, true, true, "<angle>");
-    this->DeclareOption(grpCamera, "camera-elevation-angle", "", "Camera elevation angle (in degrees)", options.CameraElevationAngle, true, true, "<angle>");
-
-    if (f3d::HasRaytracingModule)
-    {
-      auto grp5 = cxxOptions.add_options("Raytracing");
-      this->DeclareOption(grp5, "raytracing", "r", "Enable raytracing", options.Raytracing);
-      this->DeclareOption(grp5, "samples", "", "Number of samples per pixel", options.Samples, true, true, "<samples>");
-      this->DeclareOption(grp5, "denoise", "d", "Denoise the image", options.Denoise);
-    }
+    this->DeclareOption(grpCamera, "camera-position", "", "Camera position", options.getAsDoubleVectorRef("camera-position"), false, true, "<X,Y,Z>");
+    this->DeclareOption(grpCamera, "camera-focal-point", "", "Camera focal point", options.getAsDoubleVectorRef("camera-focal-point"), false, true, "<X,Y,Z>");
+    this->DeclareOption(grpCamera, "camera-view-up", "", "Camera view up", options.getAsDoubleVectorRef("camera-view-up"), false, true, "<X,Y,Z>");
+    this->DeclareOption(grpCamera, "camera-view-angle", "", "Camera view angle (non-zero, in degrees)", options.getAsDoubleRef("camera-view-angle"), false, true, "<angle>");
+    this->DeclareOption(grpCamera, "camera-azimuth-angle", "", "Camera azimuth angle (in degrees)", options.getAsDoubleRef("camera-azimuth-angle"), true, true, "<angle>");
+    this->DeclareOption(grpCamera, "camera-elevation-angle", "", "Camera elevation angle (in degrees)", options.getAsDoubleRef("camera-elevation-angle"), true, true, "<angle>");
+
+    // TODO find the right way to deal with modular options
```

Comment:
what's wrong with `if (f3d::HasRaytracingModule)`?

Reply:
Maybe just a runtime check `if (f3d::engine::hasRaytracingSupport())` would work?  
I think this public API is needed anyway.

Reply:
HasRaytracingModule is not installed. I need to think of the right way to check for the presence of optional libf3d module from the applicative side. Using the engine is fine but not great imo.

Reply:
Raytracing is a special module, it's not an extension of supported file, so it's not a problem to expose the support in the engine I think. Otherwise, you can set a public CMake variable `F3D_HAS_RAYTRACING_SUPPORT`, exported when `find_package` is called.

Reply:
>  it's not an extension of supported file, so it's not a problem to expose the support in the engine I think

It would need to be a static method, but yes. In any case, I think cmake vars would be better.

Reply:
will do in another MR. for now, lets keep the option.